tla-bmcmt/src/main/scala/at/forsyte/apalache/tla/bmcmt/rules/QuantRule.scala

```diff
@@ -214,8 +225,16 @@ class QuantRule(rewriter: SymbStateRewriter) extends RewritingRule with LazyLogg
       set: ArenaCell) = {
     val setCells = setState.arena.getHas(set)
     if (setCells.isEmpty) {
-      // \E x \in {}... is FALSE
-      setState.setRex(setState.arena.cellFalse().toNameEx)
+      // \E x \in {}: P is FALSE
+      // However, P may contain assignments inside. Bind the names to the default values.
+      val assignedNames = findAssignedNames(predEx)
+      var nextState = setState
+      for ((name, tp) <- assignedNames) {
+        nextState = defaultValueFactory.makeUpValue(nextState, CellT.fromType1(tp))
+        val newBinding = nextState.binding ++ Binding(name + "'" -> nextState.asCell)
```

Comment:
Why is this the only place where a primed name is added to the binding keys?

Reply:
The other places are handled via `AssignmentRule`